src/socket/mod.rs

```diff
@@ -0,0 +1,58 @@
+use bcc_sys::bccapi::bpf_prog_type_BPF_PROG_TYPE_SOCKET_FILTER as BPF_PROG_TYPE_SOCKET_FILTER;
+use std::collections::HashMap;
+use std::iter::Iterator;
+
+use crate::core::BPF;
+use crate::error::BccError;
+
+#[derive(Debug, Default)]
+pub struct Socket {
+    handler: Option<String>,
+    ifaces: Vec<String>,
+}
+
+impl Socket {
+    pub fn new() -> Self {
+        Default::default()
+    }
+
+    pub fn handler(mut self, handler: &str) -> Self {
+        self.handler = Some(handler.to_owned());
+        self
+    }
+
+    pub fn iface(mut self, iface: &str) -> Self {
+        self.ifaces.push(iface.to_owned());
+        self
+    }
+
+    pub fn ifaces(mut self, ifaces: &Vec<String>) -> Self {
+        self.ifaces.append(&mut ifaces.clone());
+        self
+    }
+
+    pub fn attach(self, bpf: &mut BPF) -> Result<(), BccError> {
+        if self.ifaces.len() == 0 {
+            return Err(BccError::InvalidSocket {
+                message: "iface is required".to_string(),
```

Comment:
I'd consider spelling out interface here for consistency with the other error message